pallets/gear/src/lib.rs

```diff
@@ -1677,6 +1688,59 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Pay additional rent for the program.
+        #[pallet::call_index(8)]
+        #[pallet::weight(<T as Config>::WeightInfo::pay_rent())]
+        pub fn pay_rent(
+            origin: OriginFor<T>,
+            program_id: ProgramId,
+            block_count: BlockNumberFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            ProgramStorageOf::<T>::update_active_program(
+                program_id,
+                |program| -> Result<(), Error<T>> {
+                    let block_author =
+                        Authorship::<T>::author().ok_or(Error::<T>::BlockAuthorNotFound)?;
+
+                    CurrencyOf::<T>::transfer(
+                        &who,
+                        &block_author,
+                        Self::rent_fee_for(block_count),
+                        ExistenceRequirement::AllowDeath,
+                    )
+                    .map_err(|_| Error::<T>::InsufficientBalanceForReserve)?;
+
+                    let old_block = program.expiration_block;
+                    let expiration_block = old_block.saturating_add(block_count);
+                    program.expiration_block = expiration_block;
+
+                    let task = ScheduledTask::PauseProgram(program_id);
+                    let r = TaskPoolOf::<T>::delete(old_block, task.clone());
+                    log::debug!("TaskPool::delete result = {r:?}");
+
+                    let r = TaskPoolOf::<T>::add(expiration_block, task);
+                    log::debug!("TaskPool::add result = {r:?}");
+
+                    Self::deposit_event(Event::ProgramChanged {
+                        id: program_id,
+                        change: ProgramChangeKind::ExpirationChanged {
+                            expiration: expiration_block,
+                        },
+                    });
+
+                    Ok(())
+                },
+            )
+            .map_err(|e| {
+                log::debug!("update_active_program failed: {e:?}");
```

Comment:
better to include `program_id` in log message

Reply:
ok